File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int count=0;
        ListNode* temp=head;
        while(temp){
            temp=temp->next;
            count++;
        }
        temp=head;
        int pos=count-n;
        if(pos==0){
            ListNode* curr=head->next;
            delete temp;
            return curr;
        }
        else {
            ListNode* temp=head;
            for(int i=0;i<pos-1;i++){
                temp=temp->next;
            }
            temp->next=temp->next->next;
        }
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        temp=head
        count=0
        while temp:
            temp=temp.next
            count+=1 #example=5
        if n==count:
            return head.next
        temp=head
        val=count-n #5-2=3, it means at index 3, (we want to stop 1 back)
        for i in range(val-1):
            temp=temp.next
        temp.next=temp.next.next
        return head